mantidproject/mantid
# Add Sample Geometry to Batch File

**Description of work.**
This adds sample thickness, height, and width information to the SANS batch files.
When loading a batch file, the table will be populated with the sample geometry values if there are present in the batch file, or will default to using file information if not (current behaviour).
Additionally, exporting the SANS table as a batch file will include the sample geometries

**Report to:** Steve King ISIS

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. Load the attached batch file (`Load Batch File`)
3. Click `Sample Geometry` checkbox at the bottom of the gui. This will show the sample geometry values in the table.
4. The Sample Thickness, Sample Height, Sample Width values should be 2.0, 8.0, 8.0, respectively. These are the default values taken from file information
5. Change these values in the table
6. Click the export table button (rightmost button in the row of buttons above the table) and save the table
7. Open the csv you just saved in your favourite csv reading software. The row should contain sample geometry keys with the values you input
8. Back on the SANS gui, select the row and click the remove row button
9. Then click `Load Batch File` again and select the file you saved. The sample geometries should be the values you set before exporting the table 

Fixes #25932 

**Data**:

[sans2d_batch_file.zip](https://github.com/mantidproject/mantid/files/3305223/sans2d_batch_file.zip)


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
